### DISCUSSION TITLE
Better Turbo support documentation?

### DISCUSSION THREAD

**Discussion by vfonic** [activeadmin/activeadmin#8573] [Show and tell]
I've been trying to add Turbo to ActiveAdmin v4 and it took me a while to get it working. I got it working through this Stackoverflow comment that I first found linked here on GitHub discussions:
> How to add custom javascript in Active Admin v4 ? #8346

It's the very last comment, kinda hidden in the sea of other comments. Perhaps this should be written somewhere in the docs? What do you think? Should we add this to the docs? If so, where?

Another thing is that I realized that now that I have Turbo and it's used to navigate between pages by default, when I navigate to a new page, the Flowbite JavaScript isn't reinitialized and it doesn't hook up to data attributes and it doesn't work.
The good news is that Flowbite provides support for Turbo:
https://flowbite.com/docs/getting-started/rails/
https://flowbite.com/docs/getting-started/rails/#turbo-load-support

...but it's a different exported JavaScript file. From what I can see, when using Flowbite Turbo JavaScript version, which only has some smaller changes, it doesn't work without the Turbo, so it doesn't work for ActiveAdmin initial setup. Although, we could probably replace `import "flowbite"` with:
```js
// https://flowbite.com/docs/getting-started/javascript/#init-functions
import { initFlowbite } from 'flowbite'

document.addEventListener('DOMContentLoaded', initFlowbite)
```

...or something similar.

Otherwise, we could write in the docs that, if you want to use Turbo with ActiveAdmin, you should use Flowbite Turbo and then it should work. What do you think?

I made the ActiveAdmin and Turbo (and Flowbite) work together nicely by following the Stackoverflow steps from here:
https://stackoverflow.com/a/78985377

...and adding this to my `config/importmap.rb`:

`pin "flowbite", to: "https://cdn.jsdelivr.net/npm/flowbite@2.5.2/dist/flowbite.turbo.min.js"`

**Comment by thisismydesign** [comment]
Thanks for this! It is enough to just `import 'flowbite';` in `application.js` as long as the `turbo` version is pinned.

IMO this should be handled by AA internally as it's currently not working with a default Rails app (that uses turbo). Also, otherwise, the user needs to take care to sync the flowbite version with whatever AA is using in the background.

**Reply by vfonic** [reply]
I didn't even have to do `import 'flowbite';` because the turbo version is pinned and ActiveAdmin imports flowbite for me (it uses the pinned version).

**Comment by tagliala** [comment]
> IMO this should be handled by AA internally as it's currently not working with a default Rails app (that uses turbo)

It is working with a default rails app because main `application.js` is not supposed to be loaded by active admin by default, unless I'm missing something 🤔

**Reply by thisismydesign** [reply]
Ah yes. I think this has to do with adding custom JS to AA which is discussed in https://github.com/activeadmin/activeadmin/discussions/8346

The best way I found to do it is by making AA use the default rails importmap as described here: https://stackoverflow.com/questions/78985376/activeadmin-v4-how-to-add-custom-js-via-importmaps/78985377. When I tried to extend AA's own importmap, then any changes to JS files needed a server restart to be picked up. Also in general IMO it's confusing to juggle two different setups.

**Comment by vfonic** [comment]
There's a newer discussion thread here: https://github.com/activeadmin/activeadmin/discussions/8694
